File: cpp/estrutura_sequencial/duracao.cpp
/* Problema "duracao"
 Fazer um programa para ler uma duração de tempo em segundos,
 daí imprimir na tela esta duração no formato: horas:minutos:segundos.
*/

#include <iostream>

using namespace std;

int main()
{
  int duration, hours, minutes, seconds;

  cout << "Digite a duração em segundos: ";
  cin >> duration;

  hours = duration / 3600;
  minutes = duration % 3600 / 60;
  seconds = (duration % 3600) % 60;

  cout << hours << ":" << minutes << ":" << seconds << endl;

  return 0;
}

File: csharp/Source/Estruturas/Sequencial/Duracao.cs
namespace Source.Estruturas.Sequencial;

/**
 * Problema "duracao"
 * Fazer um programa para ler uma duração de tempo em segundos,
 * daí imprimir na tela esta duração no formato: horas:minutos:segundos.
 */

internal static class Duracao
{
    public static void Execute()
    {

        Console.Write("Digite a duração em segundos: ");
        int duration = int.Parse(Console.ReadLine()!);

        int hours = duration / 3600;
        int minutes = duration % 3600 / 60;
        int seconds = (duration % 3600) % 60;

        Console.WriteLine("{0:D} : {1:D2} : {2:D2}", hours, minutes, seconds);
    }
}